src/Files.App/Helpers/DynamicDialogFactory.cs

```diff
@@ -103,6 +103,18 @@ public static DynamicDialog GetFor_RenameDialog()
 				}
 			};
 
+			inputText.TextChanged += (textBox, args) =>
+			{
+				if (!string.IsNullOrWhiteSpace(inputText.Text))
```

Comment:
Does this check for invalid characters?

Reply:
No, but as the text has already been changed we cannot revert to the old one.

Reply:
@yaira2 What should we do then?

Reply:
> No, but as the text has already been changed we cannot revert to the old one.

The best thing might be to do a little refactoring and remove the TextChanging event so the behavior is consistent. We can disable the create button if the name isn't valid and display an explanation in a TextBlock.